### PR TITLE
Update max_sub_array.py

### PR DESCRIPTION
**Pull request by SandersLin**
added another method of computing maximum sum subarray

### CODE REVIEW

File: dynamic_programming/max_sub_array.py [hunk 1]
```diff
@@ -37,7 +37,26 @@ def find_max_cross_sum(A,low,mid,high):
...
+def max_sub_array(nums: List[int]) -> int:
+    """
+     Finds the contiguous subarray (can be empty array) 
+     which has the largest sum and return its sum.
+
+    >>> max_sub_array([-2,1,-3,4,-1,2,1,-5,4])
+    6
+    >>> max_sub_array([])
+    0
+    >>> max_sub_array([-1,-2.-3])
```
Comment (cclauss) [inline dynamic_programming/max_sub_array.py]: Did you want a comma here instead of a period?

Comment (SandersLin) [reply dynamic_programming/max_sub_array.py]: yes, astonished why my local build did not catch that lol

Comment (cclauss) [reply dynamic_programming/max_sub_array.py]: I am astonished by the results of __python3 -c "print(-2.-3)"__

File: dynamic_programming/max_sub_array.py [hunk 2]
```diff
@@ -37,7 +37,26 @@ def find_max_cross_sum(A,low,mid,high):
...
+    >>> max_sub_array([])
+    0
+    >>> max_sub_array([-1,-2,-3])
+    0
+    """
+    best = 0 
+    current = 0 
+    for i in nums: 
+        current += i 
+        if current < 0: current = 0
```
Comment (cclauss) [inline dynamic_programming/max_sub_array.py]: This should be split across two lines for readability.

**Review by cclauss — CHANGES_REQUESTED** [review]

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
The good news is that this PR is merged and got us to [100 doctests](https://travis-ci.org/TheAlgorithms/Python/builds/560052564#L567) so congratulations for that.  The bad news is that this PR [raised two new pytest warnings](https://travis-ci.org/TheAlgorithms/Python/builds/560052564#L682).  Are you up for trying to fix those?

**Pull request merged**